impl/src/planning/crate_catalog.rs

```diff
@@ -167,6 +171,8 @@ impl CrateCatalogEntry {
           )
         }
       },
+      // Settings should always have `genmode` set to one of the above fields
+      GenMode::Unspecified => unreachable!(),
```

Comment:
Same thing here.

(I generally would _never_ use `unreachable` unless there's something in this function that makes that code unreachable. Having the setting be inaccessible because some far code prevents it doesn't feel very unreachable to me... or at least, it's not a very strong guarantee).